tket/src/Predicates/PassGenerators.cpp

```diff
@@ -480,14 +481,29 @@ PassPtr gen_cx_mapping_pass(
     bool delay_measures) {
   OpTypeSet gate_set = all_single_qubit_types();
   gate_set.insert(OpType::CX);
-  PassPtr rebase_pass =
-      gen_rebase_pass(gate_set, CircPool::CX(), CircPool::tk1_to_tk1);
+  PassPtr rebase_pass = gen_auto_rebase_pass(gate_set);
   PassPtr return_pass =
       rebase_pass >> gen_full_mapping_pass(arc, placement_ptr, config);
   if (delay_measures) return_pass = return_pass >> DelayMeasures();
   return_pass = return_pass >> rebase_pass >>
                 gen_decompose_routing_gates_to_cxs_pass(arc, directed_cx);
-  return return_pass;
+  Transform t{[=](Circuit& circ, std::shared_ptr<unit_bimaps_t> maps) {
```

Comment:
Does this work?
`Transform t{[=](Circuit& circ, std::shared_ptr<unit_bimaps_t> maps) {

      // Relabel all two-qubit gates
      CompilationUnit cu(circ);
      bool changed = rebase_pass->apply(cu);
      circ = cu.get_circ_ref();
      // Now try placement, falling back on "LinePlacement" if "GraphPlacement" fails
      try {
        changed = changed | placement_ptr->place(circ, maps);
      } catch (const std::runtime_error& e) {
        std::stringstream ss;
        ss << "PlacementPass failed with message: " << e.what()
          << " Fall back to LinePlacement.";
        tket_log()->warn(ss.str());
        Placement::Ptr line_placement_ptr = std::make_shared<LinePlacement>(
            placement_ptr->get_architecture_ref());
        changed = changed | line_placement_ptr->place(circ, maps);

      }
      // Now route
      MappingManager mm(std::make_shared<Architecture>(arc));
      changed = changed | mm.route_circuit_with_maps(circ, config, maps);
      
      // Now decompose routing gates to cx gates (this won't change maps)
      cu(circ);
      if(delay_measures){
        DelayMeasures()->apply(cu);
      }
      gen_decompose_routing_gates_to_cxs_pass(arc, directed_cx)->apply(cu);
      circ = cu.get_circ_ref();
      return changed;
  }`